### ISSUE TITLE
Auth

### ISSUE THREAD

**Issue by jlowin** [jlowin/fastmcp#41]
Implement auth strategies, including permissioned resources/tools/etc

**Comment by mwotton** [comment]
is there a way to do something basic like an api token? doesn't need to be fancy.

**Comment by georg-ort** [comment]
Yes please this is really missing for more professional mcp servers!

**Comment by Bschim** [comment]
by implementing #98 this should be fixed.

**Comment by jlowin** [comment]
See #362

**Comment by sakalys** [comment]
It appears that at this point the server part is covered, but the Client (sse, streamable http) do not support auth yet.

Note the MCP SDK already [supports it](https://github.com/modelcontextprotocol/python-sdk/blob/6e418e62f9446e3204f4bf889efcd9e6df76aef8/README.md#oauth-authentication-for-clients).

**Comment by jlowin** [comment]
Bearer auth (server/client) and oauth (client) are now merged to main, docs in #655. Will be released as 2.6 probably this week. More to come!
